colossus/src/main/scala/colossus/core/ServerDSL.scala

```diff
@@ -50,20 +52,18 @@ abstract class Initializer(context: InitContext) {
 
 }
 
-//almost seems like we don't need delegator anymore
-class DSLDelegator(server : ServerRef, _worker : WorkerRef, initializer: Initializer) extends Delegator(server, _worker) {
-
+object Initializer {
+  type Factory = (ServerRef, WorkerRef) => Initializer
+}
 
-  def acceptNewConnection: Option[ServerConnectionHandler] = {
-    Some(initializer.onConnect(ServerContext(server, worker.generateContext)))
-  }
+//almost seems like we don't need delegator anymore
+class DSLInitializer(server : ServerRef, _worker : WorkerRef, initializer: Initializer) extends Initializer(InitContext(server, _worker)) {
```

Comment:
This class should be removed as well.  The worker can just create a `ServerContext` itself when it calls this method.